apps/api/src/deployment/controllers/deployment/deployment.controller.ts

```diff
@@ -19,4 +28,33 @@ export class TopUpDeploymentsController {
   async cleanUpStaleDeployment(options: CleanUpStaleDeploymentsParams) {
     await this.staleDeploymentsCleanerService.cleanup(options);
   }
+
+  @Protected([{ action: "create", subject: "DeploymentSetting" }])
+  async findByDseq(dseq: string): Promise<SingleDeploymentResponse> {
+    const { currentUser } = this.authService;
+    const wallets = await this.userWalletRepository.findByUserId(currentUser.userId);
+    const deployment = await this.deploymentService.findByOwnerAndDseq(wallets[0].address, dseq);
+
+    return {
+      data: deployment
+    }
+  }
+
+  @Protected([{ action: "create", subject: "DeploymentSetting" }])
+  async create(input: CreateDeploymentRequest["data"]): Promise<CreateDeploymentResponse> {
+    let sdl: SDL;
+    try {
+      sdl = SDL.fromString(input.sdl, 'beta3');
+    } catch (_error) {
+      assert(false, 400, "Invalid SDL");
```

Comment:
should the error be checked? I mean we should ensure this is a validation issue and not some code bug in which case it should be 500.